clients/repo_client.go

```diff
@@ -53,6 +53,7 @@ type RepoClient interface {
 	ListSuccessfulWorkflowRuns(filename string) ([]WorkflowRun, error)
 	ListCheckRunsForRef(ref string) ([]CheckRun, error)
 	ListStatuses(ref string) ([]Status, error)
+	ListSboms() ([]Sbom, error)
```

Comment:
> Is an auto-generated SBOM something that is worth fetching? If it's just an API that anyone can call on any github/gitlab repo, wouldn't every repo have an SBOM then?

Thought about this last week after we talked. If what we want to track is adoption, I think fetching the auto generated boms would obviously greatly skew those numbers. Plus, for the groups that care about seeing sboms in repos, the platform generated ones are easily dismissed as not enough (at least for the time being).  I can remove the github function to fetch/check there until it's needed.

> I think given all the external discussion around SBOMs, i could see this being an interface method. But we could also potentially get this information through `ListFiles`, `ListReleases`, and `ListSuccessfulWorkflowRuns` (which may need generalized for workflow runs regardless of status).

I considered this when making the first draft. Assumed the preference was to keep things separate to allow for more configuration for the users/future dev work. I could rework this to get the info from existing methods if preferred.

Reply:
if it can be done purely as `ListFiles` and `ListReleases`, I view that as a simpler first step.

Reply:
I think `ListFiles` and `ListReleases` can replace most of the functionality, but the tricky one is `ListSuccessfulWorkflowRuns`. Since GIthub doesn't associate workflow artifacts with a workflow directly, I'd have to make an api call for each workflow returned to check for artifacts.  Unless I update it to use graphql and return more info. 

For Gitlab, the pipeline info returned by the Listcheckruns could be used to get pipeline artifacts. (The List workflowRuns might also work I'd have to double check).

Reply:
The graphql approach could be possible for fetching more data. It's also possible to just focus on source and release SBOMs first, and address CI/CD artifacts in a followup

Reply:
I think splitting the effort is making more sense. I'll pivot this MR to focus only on the release and source, and I'll create a follow on issue to add CI/CD artifact support.